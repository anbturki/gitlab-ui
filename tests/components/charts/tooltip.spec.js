import { shallowMount } from '@vue/test-utils';
import Tooltip from '../../../components/charts/tooltip/tooltip.vue';

jest.mock('echarts', () => ({
  getInstanceByDom: jest.fn(instance => instance),
}));

describe('Chart Tooltip', () => {
  let wrapper;
  const chart = {
    getDom() {
      return this;
    },
    getAttribute() {
      return '_echarts_instance_';
    },
  };

  const target = () => wrapper.find(`#${wrapper.vm.containerId}`);

  afterEach(() => {
    wrapper.destroy();
  });

  it('prevent pointer events by default', () => {
    wrapper = shallowMount(Tooltip, {
      propsData: {
        chart,
      },
    });

    expect(target().classes()).toContain('gl-pointer-events-none');
    expect(wrapper).toMatchSnapshot();
  });

  describe('allow pointer events when selectableText is true', () => {
    wrapper = shallowMount(Tooltip, {
      propsData: {
        chart,
        selectableText: true,
      },
    });

    expect(target().classes()).not.toContain('gl-pointer-events-none');
    expect(wrapper).toMatchSnapshot();
  });
});
