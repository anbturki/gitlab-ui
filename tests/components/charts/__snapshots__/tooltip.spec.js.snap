// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div id="_echarts_instance_-tooltip" class="chart-tooltip position-absolute " style="width: 1px; height: 1px;"></div>
  <gl-popover-stub cssclasses="" target="_echarts_instance_-tooltip" container="_echarts_instance_-tooltip" triggers=""><template></template> </gl-popover-stub>
</div>
`;

exports[`Chart Tooltip prevent pointer events by default 1`] = `
<div>
  <div id="_echarts_instance_-tooltip" class="chart-tooltip position-absolute  gl-pointer-events-none" style="width: 1px; height: 1px;"></div>
  <gl-popover-stub cssclasses="" target="_echarts_instance_-tooltip" container="_echarts_instance_-tooltip" triggers=""><template></template> </gl-popover-stub>
</div>
`;
